module/msg_smt/test/mod_msg_smt_unit_test.c

```diff
@@ -0,0 +1,732 @@
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Init payload buffer */
+    for (i=0; i< FAKE_MAILBOX_SIZE; i++) {
+        fake_payload[i] = i;
+    }
+
+    /* Fill payload */
+    test_reset_fake_msg();
+    fake_offset = 0;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, fake_offset, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(0, fake_msg_out[sizeof(uint32_t) + fake_offset]);
+    TEST_ASSERT_EQUAL(fake_size-1, fake_msg_out[sizeof(uint32_t) + fake_offset + fake_size - 1]);
+
+    /* Fill payload at offset */
+    test_reset_fake_msg();
+    fake_offset = 17;
+    fake_size -= fake_offset;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, fake_offset, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(0, fake_msg_out[sizeof(uint32_t) + fake_offset]);
+    TEST_ASSERT_EQUAL(fake_size-1, fake_msg_out[sizeof(uint32_t) + fake_offset + fake_size - 1]);
+}
+
+static int fake_raise_shm_notification(fwk_id_t channel_id, size_t size)
+{
+    return FWK_SUCCESS;
+}
+
+struct mod_msg_smt_driver_ouput_api fake_mbx_shm_api = {
+    .raise_notification = fake_raise_shm_notification,
+};
+
+void test_smt_respond(void)
+{
+    int i, status;
+    char fake_payload[FAKE_MAILBOX_SIZE];
+    size_t fake_size;
+    struct smt_channel_ctx *channel_ctx;
+    fwk_id_t channel_id;
+
+    /* Init scmi api */
+    channel_ctx = &smt_ctx.channel_ctx_table[FAKE_SCMI_SERVICE_IDX_OSPM_0];
+    channel_ctx->driver_api = &fake_mbx_shm_api;
+
+    /* Respond with null Payload */
+    fake_size = channel_ctx->max_payload_size;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status =  smt_respond(channel_id, NULL, fake_size);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+
+    /* No message available to write payload */
+    fake_size = channel_ctx->max_payload_size;
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, 0, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_E_ACCESS, status);
+
+    /* Signal a new message to scmi */
+    fake_signal_message_status = FWK_SUCCESS;
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = msg_signal_message(channel_id, (void *)fake_msg_in, fake_in_len, (void *)fake_msg_out, fake_out_len);
+    TEST_ASSERT_EQUAL(status, FWK_SUCCESS);
+
+     /* Init payload buffer */
+    for (i=0; i< FAKE_MAILBOX_SIZE; i++) {
```

Comment:
which one do you use ?
./tools/check_style.py looks for a checkpatch.pl. The only one that i've got is the kernel one that complains about misra c coding rules ;-D
Is there a checkpatch.pl compatible with misra c rules available somewhere ?

Reply:
That's the checkpatch.pl  that I'm using but it complains about everything including misra c coding rules

I tried to use it directly or through ./tools/check_style.py --path <path-to-checkpatch.pl>

Which parameter do you use ?

Reply:
I guess this can work ```./tools/check_style.py --path <path-to-checkpatch.pl> -m project -sl```
but if it didn't due to env. issues
eventually code_style.py calls this
```<checkpatch.pl-path>/checkpatch.pl --show-types --no-tree --no-summary --ignore LEADING_SPACE,CODE_INDENT,SUSPECT_CODE_INDENT,POINTER_LOCATION,BLOCK_COMMENT_STYLE,AVOID_EXTERNS,NEW_TYPEDEFS,VOLATILE,MACRO_WITH_FLOW_CONTROL,LINE_SPACING,SPLIT_STRING,FILE_PATH_CHANGES,PREFER_PACKED,GLOBAL_INITIALISERS,INITIALISED_STATIC --terse --file <path-to-file>```
PS: the CI didn't complain about the current commits (in terms of the style)

Reply:
the -sl helps to show spacing problem but TBH they are lost in the middle of false positive so I didn't paid much attention
also, it doesn't check all coding rules like braces {} are necessary even for single statement blocks.
Thanks for the pointer.
Side note would be good if we can find a tool that check misra c coding rules that we can run before submitting PR